[TASK] Clarify variable scope withing TCA files (#4754) (#4757)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ExtensionArchitecture/FileStructure/Configuration/TCA/Index.rst

All files in this directory are automatically included during the TYPO3
:ref:`bootstrap <bootstrapping>`.

..  versionadded:: 12.0

>     Files within :file:`Configuration/TCA/` files are loaded within a dedicated scope.

Reviewer: ```suggestion
    Files within :file:`Configuration/TCA/Overrides/` are loaded within a dedicated scope within TYPO3.
```

Author: The whole TCA directory (with Overrides subdirectory) is targetted by the described behavior.
Should I mention both directories?

Reviewer: In the TCA directory (without Overrides) you are returning an array, so there should be no leakage at all. But maybe someone defines variables there using then in the TCA definition. But that would be an edge case. So, would be okay. Thanks.

>     This means that variables defined in those files can't leak into the following files.

Reviewer: Does an average user understand this? "leak into the following files". What does that mean? A user may only have one TCA/Overrides files, so there may be no following files.

Reviewer: Sounds better for me.

>     This means that variables defined in those files can't leak into the following files.

Reviewer 2: ```suggestion
    This means that variables defined in those files can't leak to any other 
    TCA file during the TCA compilation process.
```

    ..  note::

>         Until TYPO3 v12, variables declared in these files were in a shared scope,

Reviewer: "Until TYPO3 v12" can be misunderstood. At first read, I assumed, in TYPO3 v12 the content has to be wrapped (when using local variables). But this seems not the case. So I suggest something like: "Until TYPO3 v11 and below" ..." to make it clearer.

Author: Ack

>         with the risk of a leakage to the following files. The use of :php:`call_user_func`

Reviewer: ```suggestion
        with the risk of a leakage to the following files. The use of :php:`call_user_func()`
```
